### PR TITLE
Remove `hudson.model.TreeView`

### PR DESCRIPTION
**Pull request by basil**
On December 31, 2008 (New Year's Eve), some of us were relaxing, but Kohsuke was starting to work on an experimental tree view for (what was then) Hudson in commit e666ecb18b. To this day, the class retains the warning added in the initial commit over 12 years ago:

> EXPERIMENTAL: The intention is to eventually merge this with the `ListView`. This class is here for experimentation. Until this class is sufficiently stable, there's no need for l10n.

Nothing seems to have ever used this class in core, and `usage-in-plugins` turns up no references to it in open source or CloudBees plugins. The only commits to it are codebase-wide refactorings or changes to parent classes that had to touch it in order for it to continue compiling.

Therefore it seems safe to remove this class completely rather than bothering to deprecate or restrict it.

The sole reference in a test was just looking for something that was not of type `ListView`. So I picked `MyView` instead which works equally well for this purpose.

### Proposed changelog entries

Developer: The experimental `hudson.model.TreeView` class has been removed without replacement.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by StefanSpieker — APPROVED** [review]

**Comment by daniel-beck** [comment]
Documention (sort of) at https://www.jenkins.io/doc/book/managing/system-properties/#hudson-treeview will need amending that this was removed in Jenkins 2.TBD.

Might be interesting to grab the system property from telemetry to see whether a surprisingly large number of instances actually use this, but probably not worth it.

**Review by timja — APPROVED** [review]

**Review by jtnord — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: core/src/main/resources/hudson/model/TreeView/sidepanel2_es.properties
Comment (jglick) [inline core/src/main/resources/hudson/model/TreeView/sidepanel2_es.properties:23]: > Until this class is sufficiently stable, there's no need for l10n.

so much for that

**Review by jglick — APPROVED** [review]
Has been in the back of my mind for years to kill this. Thanks!

**Review by olivergondza — APPROVED** [review]

**Comment by timja** [comment]
Seems to have caused https://github.com/jenkinsci/configuration-as-code-plugin/issues/1638, this was loading the View permission early enough for casc apparently

**Pull request merged**
